### PR TITLE
Configurable response template

### PR DESCRIPTION
**Pull request by hickey**
I have added the ability to configure how lita-jira-issues responds to a reference to a JIRA ticket. This takes the form of a configuration entry that allows the implementer to specify the format of the response with the default being what lita-jira-issues has traditionally responded with. This change has been documented in the README.md.

### CODE REVIEW

**Comment by amaltson** [comment]
Thanks for your PR, I'll take a look tonight!

File: lib/lita/handlers/jira_issues.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@ class JiraIssues < Handler
       config :password, required: true, type: String
       config :ignore, default: [], type: Array
       config :issue_ttl, default: 0, type: Integer
+      config :format, default: "[%I] %t\nStatus: %s, assigned to %a, rep. by %r, fixVersion: %v, priority: %P\n%U", type: String
```
Comment (amaltson) [inline lib/lita/handlers/jira_issues.rb]: The test is failing because the default behaviour is to leave the priority as is from JIRA. Could you please update the default priority to be `%p` and update the specs to expect lower case?

Also, have you considered a heredoc for the default example? It might read better without the need for `\n`.

File: lib/lita/handlers/jira_issues.rb [hunk 2]
```diff
@@ -34,53 +36,62 @@ def handle_key(response, key)
...
-        issue << priority(data)
-        issue << issue_link(key)
-      end
-
-      def summary(key, data)
-        "[#{key}] #{data[:summary]}"
+        
+        # build out the response from the configured format
+        text = String.new(config.format)
+        text.sub!('%I', key.upcase)
```
Comment (amaltson) [inline lib/lita/handlers/jira_issues.rb]: I like the terseness of just `%<letter>`, but I also like Ruby's idiomatic [string format using a hash](http://ruby-doc.org/core-2.2.0/String.html#method-i-25). So in this case it would look something like:

```ruby
config :format, default: "[%{I}] %{t}\nStatus: %{s}, assigned to %{a}, rep. by %{r}, fixVersion: %{v}, priority: %{P}\n%{U}", type: String
...
text = config.format % {
  I: key.upcase,
  i: key.downcase,
  S: status.upcase
...
```

You could even take it further and make it the formatter a full name like:

```ruby
FORMAT = <<-FORMATTER
[%{KEY}] %{summary}
Status: %{status}, assigned to %{assignee}, rep. by %{reporter}, fixVersion: %{fix_version}, priority: %{priority}
%{issue_link}"
  FORMATTER

config :format, default: FORMAT, type: String
```

Where an app upper case key makes the data all upper case.

**Comment by amaltson** [comment]
@hickey thank you for your PR, this looks fantastic and is infinitely more flexible than the original incarnation. Let me know what you think about the comments. Thanks!

File: lib/lita/handlers/jira_issues.rb (hunk 1 above)
Comment (hickey) [inline lib/lita/handlers/jira_issues.rb]: Yes, I finally saw the results from Travis CI. Yes, you are correct that %p would fix most of the failures. 

I have actually been thinking about how to change the substitutions for how to do uppercase, lowercase, proper case and the original value. So I will probably be changing the substitutions soon.

File: lib/lita/handlers/jira_issues.rb (hunk 2 above)
Comment (hickey) [inline lib/lita/handlers/jira_issues.rb]: Actually, I had not considered the alternative substitutions. I do sort of like using the long format of the substitutions as I can use ASSIGNEE, Assignee and assignee to select the type of case for the replacement text.

Comment (amaltson) [inline lib/lita/handlers/jira_issues.rb]: Sounds awesome :+1:. Whenever you have time to update, I'll be happy to review. Thanks again!

**Comment by hickey** [comment]
Sorry for not getting back to this earlier. Work has been kicking into overdrive and other commitments. 

I have included the changes that were discussed concerning my original code. In addition, I have added a conditional syntax that provides support for when keywords do not evaluate to a value so that the original responses can be fully supported. A bit wordy for the template, but it does allow one to provide custom text when a keyword does not have a value. 

I finally got smart and started running rspec tests locally. So sorry about spamming you earlier with the multiple TravisCI builds and failures.

**Comment by amaltson** [comment]
@hickey sorry, this fell off my radar too, a bit busy right now for an internal conference. I'll try to take a look this week. Thanks!

**Comment by amaltson** [comment]
Hey @hickey, sorry this completely fell off. I'm going to merge it as is, but I've been thinking about the templating approach and I may want to try something different (like use ERB instead of regex).

**Comment by amaltson** [comment]
@hickey I released [v0.3.0 to RubyGems](https://rubygems.org/gems/lita-jira-issues/versions/0.3.0) with the changes. You're of course [mentioned in the CHANGELOG](https://github.com/amaltson/lita-jira-issues/blob/master/CHANGELOG.md#030---2015-06-21), thanks again for your contributions!

**Comment by hickey** [comment]
Great! Thanks for the update. I can now drop the separate code base I am using and go back to using the gem!

As far as using ERB, that would make some of the templating better/cleaner and I can think of some fancy things would then be possible. Sounds like a better way to go.
